src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/CertificateRequest.cs

```diff
@@ -365,6 +365,85 @@ public X509Certificate2 CreateSelfSigned(DateTimeOffset notBefore, DateTimeOffse
             if (!issuerCertificate.HasPrivateKey)
                 throw new ArgumentException(SR.Cryptography_CertReq_IssuerRequiresPrivateKey, nameof(issuerCertificate));
 
+            DateTime notBeforeLocal = notBefore.LocalDateTime;
+            if (notBeforeLocal < issuerCertificate.NotBefore)
+            {
+                throw new ArgumentException(
+                    SR.Format(
+                        SR.Cryptography_CertReq_NotBeforeNotNested,
+                        notBeforeLocal,
+                        issuerCertificate.NotBefore),
+                    nameof(notBefore));
+            }
+
+            DateTime notAfterLocal = notAfter.LocalDateTime;
+
+            // Round down to the second, since that's the cert accuracy.
```

Comment:
So notbefore is not rounded up\down to the second? Is this actually overkill w.r.t. sub-second accuracy?